be2-scala/src/main/scala/ch/epfl/pop/pubsub/PublishSubscribe.scala

```diff
@@ -106,16 +110,27 @@ object PublishSubscribe {
           val methodPartitioner = builder.add(Partition[GraphMessage](
             totalPorts,
             {
-              case Right(m: JsonRpcRequest) if m.hasParamsMessage => portParamsWithMessage // Publish and Broadcast messages
-              case Right(m: JsonRpcRequest) if m.hasParamsChannel => portParamsWithChannel
-              case Right(_: JsonRpcRequest)                       => portParamsWithMap
-              case _                                              => portPipelineError // Pipeline error goes directly in merger
+              case Right(m: JsonRpcRequest) => m.method match {
+                  case BROADCAST          => portParamsWithMessage
+                  case PUBLISH            => portParamsWithMessage
+                  case SUBSCRIBE          => portSubscribe
+                  case UNSUBSCRIBE        => portUnsubscribe
+                  case CATCHUP            => portCatchup
+                  case HEARTBEAT          => portHeartbeat
+                  case GET_MESSAGES_BY_ID => portGetMessagesById
+                  case _                  => portPipelineError
+                }
+
+              case _ => portPipelineError // Pipeline error goes directly in merger
             }
           ))
 
           val hasMessagePartition = builder.add(ParamsWithMessageHandler.graph(messageRegistry))
-          val hasChannelPartition = builder.add(ParamsWithChannelHandler.graph(clientActorRef))
-          val hasMapPartition = builder.add(ParamsWithMapHandler.graph(dbActorRef))
+          val subscribePartition = builder.add(ParamsHandler.subscribeHandler(clientActorRef))
```

Comment:
Same suggestion here. 
**Disclaimer**: both are good, and I am not even sure if it will make the code clearer. it is just a good way to avoid having too much variables.